addons/promote/common/src/main/data/promote/rules/npm-no-pre-existing-paths.groovy

```diff
@@ -0,0 +1,26 @@
+package org.commonjava.indy.promote.rules
+
+import org.apache.commons.lang.StringUtils
+import org.commonjava.indy.promote.validate.PromotionValidationException
+import org.commonjava.indy.promote.validate.model.ValidationRequest
+import org.commonjava.indy.promote.validate.model.ValidationRule
+
+class NPMNoPreExistingPaths implements ValidationRule {
```

Comment:
I think this is no different from the maven existence check, so why not to use that one?

Reply:
If you check the maven one, you will find there is a "tools.getArtifact(path)" which is maven specified. NPM one can not apply with this.